### PR TITLE
Adds table for storing camshaft analysis nodes

### PR DESCRIPTION
**Pull request by rochoa**

### CODE REVIEW

**Comment by rochoa** [comment]
cc @javisantana @dgaubert Do you mind having a look? 🙏

**Comment by dgaubert** [comment]
👍

**Comment by rochoa** [comment]
cc @zenitraM

File: scripts-available/CDB_AnalysisCatalog.sql [hunk 1]
```diff
@@ -0,0 +1,24 @@
+-- Table to register analysis nodes from https://github.com/cartodb/camshaft
+CREATE TABLE IF NOT EXISTS
+cartodb.cdb_analysis_catalog (
+    -- md5 hex hash
+    node_id char(40) CONSTRAINT cdb_analysis_catalog_pkey PRIMARY KEY,
+    -- being json allows to do queries like analysis_def->>'type' = 'buffer'
+    analysis_def json NOT NULL,
+    -- can reference other nodes in this very same table, allowing recursive queries
+    input_nodes char(40) ARRAY NOT NULL DEFAULT '{}',
```
Comment (zenitraM) [inline scripts-available/CDB_AnalysisCatalog.sql]: would it make sense (or be possible) to add a foreign key here? so you cannot drop parent input_nodes without dropping their outputs

**Comment by zenitraM** [comment]
:+1:

File: scripts-available/CDB_AnalysisCatalog.sql (hunk 1 above)
Comment (rochoa) [inline scripts-available/CDB_AnalysisCatalog.sql]: Oh, nice suggestion! I just looked if that was possible but unfortunately postgresql does not support that yet. And I wouldn't want to create more tables for storing the relations.

**Pull request merged**